src/main/java/co/inventorsoft/academy/collections/model/Range.java

```diff
@@ -1,60 +1,273 @@
+        }
+    }
+
+    public static Range<Integer> of(Integer start, Integer end) {
+        Range<Integer> range = new Range<>();
+        if (start < end) {
+            for (Integer i = start; i <= end; i++) {
+                range.add(i);
+            }
+        }
+        return range;
+    }
+
+    public static Range<Float> of(Float start, Float end) {
+        Range<Float> range = new Range<>();
+        for (Float i = start; i <= end; i += 0.1f) {
+            range.add(i);
+        }
+        return range;
+    }
+
+    public static Range<Character> of(Character start, Character end, Function<Character, Character> func) {
+        Range<Character> range = new Range<>();
+        for (Character i = start; i.compareTo(end) <= 0; i = func.apply(i)) {
+            range.add(i);
+        }
+        return range;
+    }
+
+    public boolean add(T t) {
+        boolean flag = false;
+        if (last == null) {
+            addNode(t);
+        } else {
+            if (!contains(t)) {
+                flag = true;
+                addNode(t);
+            }
+        }
+        return flag;
+    }
+
+    private boolean addNode(T t) {
+        final Node<T> l = last;
+        final Node<T> newNode = new Node<>(l, t, null);
+        last = newNode;
+        if (l == null) {
+            first = newNode;
+        } else {
+            l.next = newNode;
+        }
+        size++;
+        return true;
+    }
 
     public int size() {
-        return 0;
+        return size;
     }
 
     public boolean isEmpty() {
-        return false;
+        return size == 0;
     }
 
     public boolean contains(Object o) {
-        return false;
+        boolean flag = false;
+        Iterator<T> iter = iterator();
+        while (iter.hasNext()) {
+            if (o.equals(iter.next())) {
+                flag = true;
+            }
+        }
+        return flag;
     }
 
+    @Override
     public Iterator<T> iterator() {
-        return null;
+        return new Iterator<T>() {
+
+            private Node<T> lastReturned;
+            private Node<T> next = first;
+            private int nextIndex = 0;
+
+            @Override
+            public boolean hasNext() {
+                return size > nextIndex;
+            }
+
+            @Override
+            public T next() {
+                if (!hasNext()) {
+                    throw new NoSuchElementException();
+                } else {
+                    lastReturned = next;
+                    next = next.next;
+                    nextIndex++;
+                }
+                return lastReturned.item;
+            }
+
+            @Override
+            public void remove() {
+                if (lastReturned == null)
+                    throw new IllegalStateException();
+
+                Node<T> lastNext = lastReturned.next;
+                if (next == lastReturned)
+                    next = lastNext;
+                else
+                    nextIndex--;
```

Comment:
Don't you have always to decrement number of elements?

Reply:
Yes you are rigth.

Reply:
Done